service/hook/bitbucketv2/bitbucketv2_test.go

```diff
@@ -976,6 +1045,39 @@ func Test_HookProvider_TransformRequest(t *testing.T) {
 		require.Equal(t, false, hookTransformResult.DontWaitForTriggerResponse)
 	}
 
+	t.Log("Test with Sample Fork Pull Request data")
+	{
+		request := http.Request{
+			Header: http.Header{
+				"X-Event-Key":      {"pullrequest:created"},
+				"Content-Type":     {"application/json"},
+				"X-Attempt-Number": {"1"},
+			},
+			Body: ioutil.NopCloser(strings.NewReader(sampleForkPullRequestData)),
+		}
+		hookTransformResult := provider.TransformRequest(&request)
+		require.NoError(t, hookTransformResult.Error)
+		require.False(t, hookTransformResult.ShouldSkip)
+		require.Equal(t, []bitriseapi.TriggerAPIParamsModel{
+			{
+				BuildParams: bitriseapi.BuildParamsModel{
+					CommitHash:               "6a3451888d91",
+					CommitMessage:            "change",
+					Branch:                   "feature/test",
+					BranchRepoOwner:          "oss-contributor",
+					BranchDest:               "master",
+					BranchDestRepoOwner:      "birmacher",
+					PullRequestID:            pointers.NewIntPtr(1),
+					BaseRepositoryURL:        "https://bitbucket.org/birmacher/prtest.git",
+					HeadRepositoryURL:        "git@bitbucket.org:oss-contributor/nice-repo.git",
```

Comment:
How come this is considered as private, but the base isn't? 🤔

Reply:
Since the provided source repository does not actually exist on Bitbucket the repo will be [handled as private.](https://github.com/bitrise-io/bitrise-webhooks/blob/steps/handle_head_base_repo_url/service/hook/bitbucketv2/bitbucketv2.go#L218-L226)